### PR TITLE
Update Thanos objsstore

### PR DESCRIPTION
**Pull request by alanprot**
**What this PR does**:
* Update to the latest thanos objectstore
* Fix 5xx when key gets revoke on store gateway
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [NA] Documentation added
- [NA] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/storage/tsdb/bucketindex/markers_bucket_client.go [hunk 1]
```diff
@@ -100,9 +100,9 @@ func (b *globalMarkersBucket) IsObjNotFoundErr(err error) bool {
 	return b.parent.IsObjNotFoundErr(err)
 }
 
-// IsCustomerManagedKeyError returns true if the permissions for key used to encrypt the object was revoked.
-func (b *globalMarkersBucket) IsCustomerManagedKeyError(err error) bool {
-	return b.parent.IsCustomerManagedKeyError(err)
+// IsAccessDeniedErr returns true if the permissions for key used to encrypt the object was revoked.
```
Comment (yeya24) [inline pkg/storage/tsdb/bucketindex/markers_bucket_client.go:103]: Do we need to update the comment?
`if the permissions for key used to encrypt the object was revoked` this seems specific to KMS error

File: pkg/storegateway/bucket_stores.go [hunk 2]
```diff
@@ -235,7 +235,7 @@ func (u *BucketStores) syncUsersBlocks(ctx context.Context, f func(context.Conte
 				if err := f(ctx, job.store); err != nil {
 					if errors.Is(err, bucket.ErrCustomerManagedKeyAccessDenied) {
 						u.storesErrorsMu.Lock()
-						u.storesErrors[job.userID] = err
+						u.storesErrors[job.userID] = httpgrpc.Errorf(int(codes.PermissionDenied), "store error: %s", err)
```
Comment (yeya24) [inline pkg/storegateway/bucket_stores.go:238]: Any reason we wrap this to httpgrpc error?

Comment (yeya24) [reply pkg/storegateway/bucket_stores.go:238]: Seems it is used in testing only?

File: pkg/storage/tsdb/bucketindex/markers_bucket_client.go (hunk 1 above)
Comment (alanprot) [reply pkg/storage/tsdb/bucketindex/markers_bucket_client.go:103]: lemme do that.

File: pkg/storegateway/bucket_stores.go (hunk 2 above)
Comment (alanprot) [reply pkg/storegateway/bucket_stores.go:238]: This get translated to a 4xx here:

https://github.com/cortexproject/cortex/blob/1d409248a9d1c1e959de5ffaad0c9070372cdc52/pkg/querier/blocks_store_queryable.go#L653-L655

This was causing 5xx in this case.

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**